frontend/src/lib/queryClient.ts

```diff
@@ -21,6 +21,19 @@ export const queryClient = new QueryClient({
     queries: {
       staleTime: 3 * 60_000,
       refetchOnWindowFocus: false,
+      retry(failureCount, error: unknown) {
+        // Do not retry for unauthorized or bad requests
+        if (
+          !!error &&
+          typeof error === 'object' &&
+          'status' in error &&
+          !isNaN(Number(error.status)) &&
+          Number(error.status) < 500
```

Comment:
What if it's a temporary 404?

Reply:
I mean you could ask that question for pretty much any status code but I think we should have some sensible defaults where we don't have a spinner for long time for requests that most likely won't succeed